pkg/error/error.go

```diff
@@ -86,72 +87,114 @@ func ConsolidateErrors(errors []ValidationError, config config.Config) []Validat
 				}
-			default:
-				// default: A human readable text format.
-				formatted_errors = append(formatted_errors, logger.LogMessage{Level: "warning", Message: fmt.Sprintf("%s:", relativeFilePath)})
-				for _, singleError := range fileErrors.Errors {
-					if singleError.LineNumber != -1 {
-						if singleError.AdditionalIdenticalErrorCount != 0 {
-							formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("\t%d-%d: %s", singleError.LineNumber, singleError.LineNumber+singleError.AdditionalIdenticalErrorCount, singleError.Message)})
-						} else {
-							formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("\t%d: %s", singleError.LineNumber, singleError.Message)})
-						}
+			}
+		}
+	}
+
+	return formatted_errors
+}
+
+func FormatErrorsAsGHA(errors []ValidationErrors, config config.Config) []logger.LogMessage {
+	var formatted_errors []logger.LogMessage
+
+	for _, fileErrors := range errors {
+		if len(fileErrors.Errors) > 0 {
+			relativeFilePath, err := files.GetRelativePath(fileErrors.FilePath)
+			if err != nil {
+				formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: err.Error()})
+				continue
+			}
+
+			fileErrors.Errors = ConsolidateErrors(fileErrors.Errors, config)
+
+			// github-actions: A format dedicated for usage in Github Actions
+			for _, singleError := range fileErrors.Errors {
+				if singleError.LineNumber != -1 {
+					if singleError.AdditionalIdenticalErrorCount != 0 {
+						formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("::error file=%s,line=%d,endLine=%d::%s", relativeFilePath, singleError.LineNumber, singleError.LineNumber+singleError.AdditionalIdenticalErrorCount, singleError.Message)})
 					} else {
-						formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("\t%s", singleError.Message)})
+						formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("::error file=%s,line=%d::%s", relativeFilePath, singleError.LineNumber, singleError.Message)})
 					}
+				} else {
+					formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: fmt.Sprintf("::error file=%s::%s", relativeFilePath, singleError.Message)})
 				}
 			}
 		}
 	}
 
+	return formatted_errors
+}
+
+// gcc: A format mimicking the error format from GCC.
+func FormatErrorsAsGCC(errors []ValidationErrors, config config.Config) []logger.LogMessage {
+	var formatted_errors []logger.LogMessage
+
+	for _, fileErrors := range errors {
+		if len(fileErrors.Errors) > 0 {
+			relativeFilePath, err := files.GetRelativePath(fileErrors.FilePath)
+			if err != nil {
+				formatted_errors = append(formatted_errors, logger.LogMessage{Level: "error", Message: err.Error()})
+				continue
+			}
+
+			for _, singleError := range fileErrors.Errors {
+				var lineNo = 0
```

Comment:
Idiomatic way would be either

```suggestion
				lineNo := 0
```
or

```suggestion
				var lineNo int
```

Reply:
done